README.rst

```diff
@@ -194,7 +192,9 @@ This backend can be run in serial or blocked mode and defaults to running in the
 if ``/cpu/self/memcheck`` is selected at runtime.
 
 The ``/*/occa`` backends rely upon the `OCCA <http://github.com/libocca/occa>`_ package to provide
-cross platform performance.
+cross platform performance. Additionally, users can pass specific device flags such as:
+
+  - `"/*/occa/{ mode: 'CUDA', device_id: 0 }"`
```

Comment:
@jedbrown, since this is a pattern that would be adopted by other backends, we probably want to make an intentional decision on the syntax here. I recall you having an opinion.

Reply:
Could it just be like `/gpu/cuda/occa/0`? For non-hierarchical options, we could consider URL notation, `/gpu/cuda/occa?key1=val1?key2=val2` or with other separators such as `/gpu/cuda/occa,key1=val1,key2=val2`.

Reply:
Copying the comment above:

> I like the original design with
> ```
> /<hardware>/<backend>
> ```
> Even though OCCA reuses a backend implementation, I don't think we should nest them like
> 
> ```
> /gpu/cuda/occa
> ```

I do like the idea of standardizing the arguments. if we're going for a URI-type format, I like 
```
/gpu/occa?mode=OpenCL&platform_id=0&device_id=0
```

Reply:
Using `?` and `&` as delimiters is bad for readability. For me, and probably many other people, it is much easier to parse `/gpu/occa:mode=OpenCL,platform_id=0,device_id=0`.

Reply:
A benefit of using `:` and/or `,` is that they aren't shell magic and thus don't need to be quoted.